internals/updates/update.go

```diff
@@ -0,0 +1,40 @@
+package updates
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+type GhostUpdaterCleaner interface {
+	Clean(ctx context.Context) error
+}
+
+func Remove(ctx context.Context, updateID int, fn GhostUpdaterCleaner) error {
+	// Make three attempts to allow the updater container to exit on its own.
+	// This is important because if the updater container is forcibly removed
+	// before it exits naturally, it may skip removing the previous agent's
+	// container by the updater container, resulting in a container name conflict
+	// during the next remote update.
```

Comment:
what about just like in docker, check if the updater job is done, if it's not done, continue and wait more?

Reply:
When using Docker, we need to rename the new container manually once the updater job removes the previous one. This is because there is always a container with a unique name, `portainer_agent`. However, with Nomad as an orchestrator, the old container is automatically removed after the new container is created. There is no unique container name, but instead a random string is appended to the name `portainer_agent`, resulting in `portainer_agent_xxxxxx`.